osf/models/preprint_provider.py

```diff
@@ -70,6 +70,10 @@ def __unicode__(self):
         return '{} with id {}'.format(self.name, self.id)
 
     @property
+    def has_custom_subjects(self):
+        return self.subjects.exists()
```

Comment:
Based on the linked EOSF ticket, this looks like it should be self.subjects.filter(highlighted=True).exists() and change https://github.com/CenterForOpenScience/osf.io/pull/7883/files#diff-03133408b78d570d4fe79642d38b06c8L74 to use this new property.